testsuite/k8s-conformance-test-suite.sh

```diff
@@ -1,68 +1,142 @@
-# kubectl create secret generic sonobuoy-proxy-cert --from-file=proxycert=<path-to-cert-file>
-# By default we check for the secret in the default namespace. In case you have created the secret in some other namespace, please add the following variables in the sonobuoy run command: 
-# --plugin-env azure-arc-platform.PROXY_CERT_NAMESPACE="<namespace of sonobuoy secret>"
-# --plugin-env azure-arc-agent-cleanup.PROXY_CERT_NAMESPACE="namespace of sonobuoy secret"
+# read config file i.e., azure-arc-conformance.properties
+declare -A properties
+declare -A enabled_plugins
+
+properties_count=0
+plugin_count=0
+repository="https://github.com/santosh02iiit/azure-arc-validation.git"
+
+while IFS= read -r line; do
+    # remove leading and trailing whitespaces
+    line="$(echo -e "${line}" | sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//')"
+
+    if [[ "${line}" == "#"* ]] #ignore lines starting with comments
+    then
+        continue
+    elif [ -z "${line}" ] #ignore empty lines
+    then
+        continue
+    fi
+
+    if [[ $line == *"enable"* ]] 
+    then
+        if [[ $line == *"true"* ]]
+        then
+            enabled_plugins[$plugin_count]=$(echo $line | cut -d. -f1)
+            ((plugin_count++))
+        fi
+    
+    elif [[ $line == *"dev-repository"* ]]
+    then
+        repository=$(echo $line | cut -d= -f2)
+
+    elif [[ $line == *"offering-name"* ]]
+    then
+        OFFERING_NAME=$(echo $line | cut -d= -f2)
+
+    elif [[ $line == *"az-storage-account-sas"* ]]
+    then
+        AZ_STORAGE_ACCOUNT_SAS="${line#*=}"
+        echo "AZ_STORAGE_ACCOUNT_SAS : $AZ_STORAGE_ACCOUNT_SAS"
+    
+    elif [[ $line == *"az-storage-account"* ]]
+    then
+        AZ_STORAGE_ACCOUNT=$(echo $line | cut -d= -f2)
+
+    else
+        properties[$properties_count]=$line
+        ((properties_count++))
+    fi
+
+    if [[ $line == *"SUBSCRIPTION_ID"* ]]
+    then
+        AZ_SUBSCRIPTION_ID="${line#*=}"
+
+    elif [[ $line == *"CLIENT_ID"* ]]
+    then
+        AZ_CLIENT_ID="${line#*=}"
+
+    elif [[ $line == *"CLIENT_SECRET"* ]]
+    then
+        AZ_CLIENT_SECRET="${line#*=}"
+    
+    elif [[ $line == *"TENANT_ID"* ]]
+    then
+        AZ_TENANT_ID="${line#*=}"
+    fi
+
+done < /etc/config/azure-arc-conformance.properties
 
 az login --service-principal --username $AZ_CLIENT_ID --password $AZ_CLIENT_SECRET --tenant $AZ_TENANT_ID
 az account set -s $AZ_SUBSCRIPTION_ID
 
+# Url will be changed test code
+echo "arc repository: $repository"
+git clone $repository
+cd /azure-arc-validation
+git checkout -b launcher_VMwareProposal origin/launcher_VMwareProposal
+cd /
+# test code ends
+
+plugins_file=(/azure-arc-validation/testsuite/conformance-test-plugins/*)
+
+command="sonobuoy run --wait"
+#check if yaml file is present for the user configuration
+add_helm=0
+found=0
+for enabled_plugin in "${enabled_plugins[@]}"
+do
+    for file in "${plugins_file[@]}"
```

Comment:
We can skip iterating over the extension plugin yamls. We can simply mandate that the plugin yaml should have standard naming convention as '<plugin_name>.yaml'. So we can directly append the '--plugin <path_to_plugin_yaml>' part just by knowing the plugin name

Reply:
Its true, but i wanted to check if the file is present in the directory for the enabled plugin (yes and that should be always present). This also handle case when the  enabled plugin is misspelled  we raise an error message, if we remove this sonobuoy will need to handle this.

Let me know your thoughts

Reply:
In current code we are throwing error message on below cases:
1. plugin yaml file is not present in the repo
2. plugin name is misspelled
In both the cases the plugin will be ignored.